Docs/developer_guide/script_repository.rst

```diff
@@ -0,0 +1,4237 @@
+     if isinstance(node, slicer.vtkMRMLVolumeNode):
+       # Call showVolumeRendering using a timer instead of calling it directly
+       # to allow the volume loading to fully complete.
+       qt.QTimer.singleShot(0, lambda: showVolumeRendering(node))
+
+   def showVolumeRendering(volumeNode):
+     print("Show volume rendering of node "+volumeNode.GetName())
+     volRenLogic = slicer.modules.volumerendering.logic()
+     displayNode = volRenLogic.CreateDefaultVolumeRenderingNodes(volumeNode)
+     displayNode.SetVisibility(True)
+     scalarRange = volumeNode.GetImageData().GetScalarRange()
+     if scalarRange[1]-scalarRange[0] < 1500:
+       # small dynamic range, probably MRI
+       displayNode.GetVolumePropertyNode().Copy(volRenLogic.GetPresetByName('MR-Default'))
+     else:
+       # larger dynamic range, probably CT
+       displayNode.GetVolumePropertyNode().Copy(volRenLogic.GetPresetByName('CT-Chest-Contrast-Enhanced'))
+
+   slicer.mrmlScene.AddObserver(slicer.vtkMRMLScene.NodeAddedEvent, onNodeAdded)
+
+Automatically load volumes that are copied into a folder
+--------------------------------------------------------
+
+This example shows how to implement a simple background task by using a timer. The background task is to check for any new volume files in folder and if there is any then automatically load it.
+
+There are more efficient methods for file system monitoring or exchanging image data in real-time (for example, using OpenIGTLink), the example below is just for demonstration purposes.
+
+.. code-block:: python
+
+   incomingVolumeFolder = "c:/tmp/incoming"
+   incomingVolumesProcessed = []
+
+   def checkForNewVolumes():
+     # Check if there is a new file in the
+     from os import listdir
+     from os.path import isfile, join
+     for f in listdir(incomingVolumeFolder):
+       if f in incomingVolumesProcessed:
+         # this is an incoming file, it was already there
+         continue
+       filePath = join(incomingVolumeFolder, f)
+       if not isfile(filePath):
+         # ignore directories
+         continue
+       logging.info("Loading new file: "+f)
+       incomingVolumesProcessed.append(f)
+       slicer.util.loadVolume(filePath)
+     # Check again in 3000ms
+     qt.QTimer.singleShot(3000, checkForNewVolumes)
+
+   # Start monitoring
+   checkForNewVolumes()
+
+DICOM
+-----
+
+How to load DICOM files into the scene from a folder
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This code loads all DICOM objects into the scene from a file folder. All the registered plugins are evaluated and the one with the highest confidence will be used to load the data. Files are imported into a temporary DICOM database, so the current Slicer DICOM database is not impacted.
+
+.. code-block:: python
+
+   dicomDataDir = "c:/my/folder/with/dicom-files"  # input folder with DICOM files
+   loadedNodeIDs = []  # this list will contain the list of all loaded node IDs
+
+   from DICOMLib import DICOMUtils
```

Comment:
To suggest best practices, I am wondering if we should move the import at the top